From a28406a6783bfd726c1972339076d2a2233bc72b Mon Sep 17 00:00:00 2001
From: Juliusz Sosinowicz <juliusz@wolfssl.com>
Date: Thu, 4 Apr 2024 20:16:15 +0200
Subject: [PATCH] tests: Tune ap_wpa2_eap_fast_prf_oom for wolfssl

The wolfSSL backend only does one malloc in
tls_connection_get_eap_fast_key(). Failing on the second one skips the
only malloc and fails the test.

Signed-off-by: Juliusz Sosinowicz <juliusz@wolfssl.com>
---
 tests/hwsim/test_ap_eap.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/tests/hwsim/test_ap_eap.py b/tests/hwsim/test_ap_eap.py
index 6cf427eb1..8dddbf75b 100644
--- a/tests/hwsim/test_ap_eap.py
+++ b/tests/hwsim/test_ap_eap.py
@@ -4247,9 +4247,12 @@ def test_ap_wpa2_eap_fast_prf_oom(dev, apdev):
     """WPA2-Enterprise connection using EAP-FAST and OOM in PRF"""
     check_eap_capa(dev[0], "FAST")
     tls = dev[0].request("GET tls_library")
-    if tls.startswith("OpenSSL") or tls.startswith("wolfSSL"):
+    if tls.startswith("OpenSSL"):
         func = "tls_connection_get_eap_fast_key"
         count = 2
+    elif tls.startswith("wolfSSL"):
+        func = "tls_connection_get_eap_fast_key"
+        count = 1
     elif tls.startswith("internal"):
         func = "tls_connection_prf"
         count = 1
-- 
2.43.0

